### PR TITLE
PSR-604 Added shares list page

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

File: app/views/TwoColumnsTripleAction.scala.html [hunk 1]
```diff
@@ -66,12 +60,16 @@ <h1 class="govuk-heading-l">@renderMessage(viewmodel.heading)</h1>
         @viewmodel.page.paginatedViewModel.map{ paginatedViewModel =>
             @if(paginatedViewModel.pagination.totalPages > 1) {
                 <p id="pagination-label" class="govuk-body">@{renderMessage(paginatedViewModel.label)}</p>
+                @govukPagination(PaginationViewModel(paginatedViewModel.pagination))
             }
-            @govukPagination(PaginationViewModel(paginatedViewModel.pagination))
```
Comment (westwater) [inline app/views/TwoColumnsTripleAction.scala.html:70]: bug fix, was creating an empty div and applying margin when left out of braces

File: app/controllers/nonsipp/shares/SharesListController.scala [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  psrSubmissionService: PsrSubmissionService,
+  saveService: SaveService
+)(implicit ec: ExecutionContext)
+    extends PSRController {
+
+  val form: Form[Boolean] = SharesListController.form(formProvider)
+
+  def onPageLoad(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
+    implicit request =>
+      // remove once last page is developed
```
Comment (kberec) [inline app/controllers/nonsipp/shares/SharesListController.scala:64]: Page 26r is now done, is this the last page you meant @westwater ?

File: app/pages/nonsipp/shares/SharesCompleted.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import models.SchemeId.Srn
+import pages.{IndexedQuestionPage, QuestionPage}
+import play.api.libs.json.JsPath
+import utils.RefinedUtils.RefinedIntOps
+import viewmodels.models.SectionCompleted
+
+case class SharesCompleted(srn: Srn, index: Max5000) extends QuestionPage[SectionCompleted] {
+
+  override def path: JsPath =
+    Paths.shareTransactions \ toString \ index.arrayIndex.toString
```
Comment (kberec) [inline app/pages/nonsipp/shares/SharesCompleted.scala:29]: Just a question really as a matter of interest. Other 'completed' pages, e.g. `EmployerContributionsCompleted` are having a path set up to be on root, e.g. `override def path: JsPath = JsPath \ ......`
This is now relative to shareTransactions. Any specific reasons to have it there?

**Review by kberec — COMMENTED** [review]
Looks good.

File: app/pages/nonsipp/shares/SharesCompleted.scala (hunk 3 above)
Comment (westwater) [reply app/pages/nonsipp/shares/SharesCompleted.scala:29]: Just easier to see when you want to look at the raw data (completed shares data next to other shares data) - must have been an oversight with the others but as it doesn't get sent to ETMP it won't be of any consequence

File: app/controllers/nonsipp/shares/SharesListController.scala (hunk 2 above)
Comment (westwater) [reply app/controllers/nonsipp/shares/SharesListController.scala:64]: Yeah I think so, I'll raise another PR after this to quickly stitch everything together (don't want to wait too long on this PR incase of another conflict)

**Review by kberec — APPROVED** [review]

**Pull request merged**